RFC-0004-sparse-fill-value/README.md

```diff
@@ -0,0 +1,613 @@
+
+intracellular_concentration_of_potassium_var = torch.sparse_coo_tensor(
+    indices = signal.indices(),
+    values = intracellular_concentration_of_potassium_var_values,
+    size = signal.size())
+```
+
+which is convoluted, error-prone, and does not result in a single data
+structure as is `intracellular_concentration_of_potassium` above.
+
+## Review of sparse array software
+
+As a rule, software that implements sparse matrix support does not
+implement support for nonzero fill value. The reason for this is
+twofold: either the application of the software do not require this
+feature (such as applications from Linear Algebra and Graph theory),
+or the feature is declared 'not implemented yet'. However, software
+also exists that implements the nonzero fill value support for sparse
+matrices.  In the following, we will review a selection of sparse
+array software to learn the different approaches to handling the fill
+value feature (or the lack of it).
+
+### Wolfram Language
+
+[SparseArray](https://reference.wolfram.com/language/ref/SparseArray.html)
+has a constructor variant `SparseArray[data,dims,val]` that "yields a
+sparse array in which unspecified elements are taken to have value
+`val`". The default fill value is 0.
+
+### Pandas - Python Data Analysis Library
+
+Pandas
+[SparseArray](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.arrays.SparseArray.html#pandas.arrays.SparseArray)
+supports `fill_value` option to enable memory saving for most common
+value in an array. The default fill value depends on array dtype, for
+instance, it is `nan` for `float` and `0` for `int`. The
+interpretation of the default fill value is "missing data".
+
+### MathWorks
+
+Matlab's [sparse
+matrix](https://www.mathworks.com/help/matlab/ref/sparse.html) does
+not support nonzero fill value feature. Historically, the idea of
+nonzero fill value was rejected in the interest of simplicity, [see
+Sec 2.5 in this
+paper](https://epubs.siam.org/doi/abs/10.1137/0613024).
+
+### Scikit-Learn/SciPy.sparse
+
+[Scikit-Learn defines two
+semantics](https://scikit-learn.org/stable/glossary.html#term-sparse-matrix)
+for interpreting the unspecified entries in a sparse matrix:
+
+- Matrix (here Linear Algebra) semantics where the fill value is
+  assumed to be 0,
+- Graph semantics where the fill value is assumed to be indefinite
+  value.
+
+The Scikit-Learn algorithms of the Matrix and Graph domains use sparse
+arrays from
+[scipy.sparse](https://docs.scipy.org/doc/scipy/reference/sparse.html)
+which do not record the fill value in the sparse array object
+explicitly. Instead, the corresponding algorithms make an implicit
+assumption about the fill value when processing the same SciPy sparse
+arrays.
+
+See also [the feature request of default value for scipy.sparse](https://stackoverflow.com/questions/6256206/scipy-sparse-default-value).
```

Comment:
Is it possible to add `PyData/Sparse` here? 😉

Reply:
Yes. Actually, the reference to PyData/Sparse replaces the stackoverflow link.